Dockerfile

```diff
@@ -19,10 +24,54 @@ RUN yum -y install \
     freetype-devel \
     libart_lgpl-devel \
     gcc \
-    groff
+    groff \
+    wget \
+    automake \
+    libffi-devel \
+    libyaml-devel
+
+# RUN cd tmp && \
+#     wget https://www.openssl.org/source/openssl-1.0.2a.tar.gz && \
+#     tar -zxvf openssl-*.tar.gz && \
+#     cd openssl-*/ && \
+#     ./config -fpic shared && make && make install && \
+#     echo "/usr/local/ssl/lib" >> /etc/ld.so.conf && \
+#     ldconfig
+
+RUN yum -y install \
+    make \
+    automake \
+    autoconf \
+    install \
+    perl-ExtUtils-MakeMaker \
+    fakeroot
+
+RUN curl -o /tmp/openssl-1.0.2u.tar.gz http://artfiles.org/openssl.org/source/old/1.0.2/openssl-1.0.2u.tar.gz && \
+    cd /tmp && tar -xzf /tmp/openssl-1.0.2u.tar.gz && \
+    cd /tmp/openssl-1.0.2u && ./Configure linux-x86_64 no-shared --openssldir=/opt/openssl -fPIC && make && make install && \
+    echo "/usr/local/ssl/lib" >> /etc/ld.so.conf && ldconfig && \
+    cd - && rm -rf /tmp/openssl-1.0.2u && rm /tmp/openssl-1.0.2u.tar.gz
+
+COPY ./curl-7.46.0.tar.gz /tmp/curl-7.46.0.tar.gz
+
+RUN cd /tmp && tar -xzf /tmp/curl-7.46.0.tar.gz && \
+    cd /tmp/curl-7.46.0 && LIBS="-ldl" ./configure --enable-static --prefix=/opt/curl --with-ssl=/opt/openssl && make all && make install && \
+    cd - && rm -rf /tmp/curl-7.46.0 && rm -f /tmp/curl-7.46.0.tar.gz
+
+# # Upgrade openssl
+# RUN rpm -Uvh  http://repoforge.eecs.wsu.edu/redhat/el5/en/x86_64/rpmforge/RPMS/$(curl -s http://repoforge.eecs.wsu.edu/redhat/el5/en/x86_64/rpmforge/RPMS/ | grep rpmforge-release | grep x86_64 | grep el5 | sort | tail -n1 | sed 's%.*>\(rpmforge\-release\-.*.rpm\)<.*%\1%')
+# RUN sed -i 's/apt\.sw\.be/repoforge.eecs.wsu.edu/g' /etc/yum.repos.d/rpmforge.repo && \
+#     sed -i '/rpmforge-extras/,/^enabled\|^\[/s/^enabled.*/enabled = 1/' /etc/yum.repos.d/rpmforge.repo
+#
+# RUN yum -y install \
+#     install \
+#     perl-ExtUtils-MakeMaker \
+#     fakeroot
+
+ENV PATH="/opt/curl/bin:${PATH}"
 
 # Set up an RVM with Ruby 2.2.2
-RUN gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+RUN gpg --keyserver hkp://pool.sks-keyservers.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
```

Comment:
I'm guessing the old key isn't valid anymore - do we need to download it?

Reply:
I don't want to remove anything from the image that I don't have to. This should be innocuous.